ihmc-perception/src/main/java/us/ihmc/perception/RawImagePublisher.java

```diff
@@ -102,6 +110,32 @@ private void publishAsImageMessage(ROS2Topic<ImageMessage> imageTopic, RawImage
       ros2Helper.publish(imageTopic, imageMessage);
    }
 
+   private void publishAsROS2CompatImage(ROS2Topic<ROS2CompatImage> imageTopic, RawImage imageToPublish)
+   {
+      ROS2CompatImage compatImage = new ROS2CompatImage();
+
+      compatImage.getHeader().getStamp().setSec((int) imageToPublish.getAcquisitionTime().getEpochSecond());
+      compatImage.getHeader().getStamp().setNanosec(imageToPublish.getAcquisitionTime().getNano());
+      compatImage.getHeader().setFrameId(Long.toString(imageToPublish.getSequenceNumber()));
+
+      compatImage.setHeight(compatImage.getHeight());
+      compatImage.setWidth(compatImage.getWidth());
+
+      PixelFormat pixelFormat = PixelFormat.fromImageMessage(imageMessage);
+      System.out.println("pixelFormat " + pixelFormat.name());
+      // TODO: set encoding
+
+      compatImage.setIsBigendian((byte) 0);
+//      compatImage.setStep(0);
```

Comment:
Pretty sure this'll be: 
```suggestion
      compatImage.setStep(imageToPublish.getCpuImageMat().step());
```